src/boehm_gc.c

```diff
@@ -356,7 +369,7 @@ void RetypeBagIntern(Bag bag, UInt new_type)
             old_mem = PTR_BAG(bag);
             old_mem = ((char *)old_mem) - sizeof(BagHeader);
             memcpy(new_mem, old_mem, size);
-            SET_PTR_BAG(bag, (void *)(((char *)new_mem) + sizeof(BagHeader)));
+            SET_PTR_BAG(bag, DATA(new_mem));
```

Comment:
I don't quite understand why `DATA(new_mem)` and `(void *)(((char *)new_mem) + sizeof(BagHeader))` to the same thing.

Reply:
The definition of `DATA` is
```C
static inline Bag * DATA(BagHeader * bag)
{
    return (Bag *)(((char *)bag) + sizeof(BagHeader));
}
```
hence `DATA(new_mem)` evaluates to almost the exact same expression, the only difference is that the final result is case to `Bag *` instead of `void *`, but since `SET_PTR_BAG` expects a `Bag *` as its second argument, the `void *` was implicitly converted to a `Bag *` anyway.

Does that clear it up?

Reply:
Yes thanks! I was erroneously looking at 
```C
static inline Bag *DATA(BagHeader *bag)
{
    return (Bag *)(bag + 1);
}
```
in `gasman.c`.

Reply:
That's actually equivalent, too ;-). In C, if you have a pointer at a type (such as `BagHeader`), then adding 1 to that pointer increases the pointer (viewed as a "byte address") by `sizeof(ThatType)`